src/main/java/org/jacop/constraints/XneqY.java

```diff
@@ -105,6 +110,11 @@ public XneqY(IntVar x, IntVar y) {
 
     @Override public void notConsistency(final Store store) {
 
+        if (x == y) {
+            // If x and y are the same, there is trivially nothing to do.
+            return;
```

Comment:
Models that impose constraints like 
store.impose(new Not(new XneqY(x, x)))
I would consider as bad coding practice as it may have also bad effect on search heuristics (if for example 
MostConstrained variable selection is used) and efficiency. 
However, it is better if we can handle this weird type of models more efficiently. 
Thanks for raising this issue.

Reply:
I agree that it would be inefficient to do this in the first place, but as the interface does not prevent you from modelling constraints in the form of `new Not(new XneqY(x, x))` I think we should keep them in mind regardless. But, primarily, I added this so that `consistency()` and `notConsistency()` remain similar in structure.
Thanks!